admin/cleanup.py

```diff
@@ -0,0 +1,379 @@
+        older than the maximum age.
+
+        :param timedelta maximum_age: The oldest a volume may be without being
+            considered old enough to include in the result.
+        :param list all_volumes: The libcloud volumes representing all the
+            volumes we can see on a particular cloud service.
+
+        :rtype: ``VolumeActions``
+        """
+        now = datetime.now(tz=tzutc())
+        destroy = []
+        keep = []
+        for volume in volumes:
+            created = _get_volume_creation_time(volume)
+            if self._is_test_volume(volume) and now - created > maximum_age:
+                destroy.append(volume)
+            else:
+                keep.append(volume)
+        return VolumeActions(destroy=destroy, keep=keep)
+
+    def _destroy_cloud_volumes(self, volumes):
+        """
+        Unconditionally and irrevocably destroy all of the given cloud volumes.
+        """
+        for volume in volumes:
+            try:
+                volume.destroy()
+            except:
+                err(None, "Destroying volume.")
+
+
+def _dumps(obj):
+    """
+    JSON encode an object using some visually pleasing formatting.
+    """
+    return json.dumps(obj, sort_keys=True, indent=4, separators=(',', ': '))
+
+
+def _format_time(when):
+    """
+    Format a time to ISO8601 format.  Or if there is no time, just return
+    ``None``.
+    """
+    if when is None:
+        return None
+    return datetime.isoformat(when)
+
+
+def _get_tag(volume, tag_name):
+    """
+    Get a "tag" from an EBS or Rackspace volume.
+
+    libcloud doesn't represent tags uniformly across drivers.  Thus this method
+    only works on drivers specifically account for.
+
+    :raise: ``KeyError`` if the tag is not present.
+    """
+    return volume.extra.get("tags", volume.extra.get("metadata"))[tag_name]
+
+
+@attributes(["destroy", "keep"])
+class VolumeActions(object):
+    """
+    Represent something to be done to some volumes.
+
+    :ivar destroy: Volumes to destroy.
+    :ivar keep: Volumes to keep.
+    """
+
+
+def _existing_file_path_option(option_name, option_value):
+    file_path = FilePath(option_value)
+    if not file_path.exists():
+        raise UsageError(
+            u"Problem with --{}. File does not exist: '{}'.".format(
+                option_name, file_path.path
+            )
+        )
+    return file_path
+
+
+def _yaml_configuration_path_option(option_name, option_value):
+    yaml_path = _existing_file_path_option(option_name, option_value)
+    try:
+        configuration = yaml.safe_load(yaml_path.open())
+    except ParserError as e:
+        raise UsageError(
+            u"Problem with --{}. "
+            u"Unable to parse YAML from {}. "
+            u"Error message: {}.".format(
+                option_name, yaml_path.path, unicode(e)
+            )
+        )
+    return configuration
+
+
+class CleanupCloudResourcesOptions(Options):
+    """
```

Comment:
nit: empty docstring. Is this some weird quirk about `Options` classes, or should this be filled with goodies or removed?